src/Integration/State/StateManager.cs

```diff
@@ -127,8 +129,15 @@ private void OnIsBusyChanged(bool isBusy)
             this.IsBusyChanged?.Invoke(this, isBusy);
         }
 
-        private void OnBindingStateChanged()
+        private void OnBindingStateChanged(bool isBound)
         {
+            if (this.isSolutionBound == isBound)
```

Comment:
Instead you should change the event to return this result i.e. use EventHandler<bool>.
The handler need to decide how to respond to this rather than the sender.